### DISCUSSION TITLE
Allow pointing to existing dev server assets

### DISCUSSION THREAD

**Discussion by plunkettscott** [wailsapp/wails#1095] [Ideas]
It seems like Wails 2 could support the ability to point to an external dev server (such as Vite) during local development and to gain HMR support. The watch/rebuild process we have to do today with SvelteKit (there is no plan to watch builds with SvelteKit) kills some of the productivity gained from modern web frameworks. In similar projects, like [Tauri](https://tauri.studio) (Rust), they allow use of an external dev server which leads to a more productive experience because build time is basically instant when only the frontend code changes (thanks to Vite).

Is this something we could consider instead of having to compile assets and serve them from the Wails dev server? Maybe Wails dev server (if required for Wails v2 to function) could proxy requests to the external dev server?

**Comment by leaanthony** [comment]
Hey there. Yes, it's been discussed many times and there *is* the `devserverurl` flag for this purpose. The thing is, I'm not convinced it works 100% and nobody has really taken it on to get it fixed. If you have experience with other frameworks doing this then maybe you are best placed to define how it should work and how we might fix what we currently have? I've always said I'm willing to give a good chunk of time to this - it just needs someone to work with it and test it.

**Reply by leaanthony** [reply]
I should also say, there's a couple of community template using vite though I've never tested them: https://wails.io/docs/community/templates

**Reply by plunkettscott** [reply]
Yeah, I definitely thought that was what the intended use of `devserverurl` was, but it seems to still serve the assets from the `assetdir` path. Let me look through the codebase and see if I may be able to tackle this and understand the current issues.

**Comment by plunkettscott** [comment]
@leaanthony, looking through the codebase, I see this:

https://github.com/wailsapp/wails/blob/9d519a3a3fc54c1d127c06e73554c932d393fb0d/v2/internal/frontend/devserver/devserver.go#L105

But I don't see anything that actually proxies the devserverurl when it's not the default. Am I missing where this happens?

**Reply by leaanthony** [reply]
I'll check today. This thread has more info on this https://github.com/wailsapp/wails/discussions/828#discussioncomment-1419485

**Reply by plunkettscott** [reply]
Thanks! I think it just might not be doing what I expect it to be doing, which is fine. I'm just having a hard time connecting all the pieces that make up this feature.
